Add a bit of extra info about packet ordering (#191)
memfault/memfault-docs: docs/mcu/data-from-firmware-to-the-cloud.mdx

  [chunks HTTP API](https://api-docs.memfault.com/) sequentially, in the order
  in which the Firmware SDK's packetizer created the chunks. When posting chunks
  concurrently, ensure that requests for _the same device_ cannot happen

>   concurrently, to avoid violating the ordering requirement. Out of order chunks
>   will be best-effort reordered on the server, but the **first** and **last**
>   chunks MUST be received in the correct position for decoding to succeed.

Reviewer: ```suggestion
  concurrently, to avoid violating the ordering requirement. Out of order chunks
  will be best-effort reordered on the Memfault server.
```
